src/test/java/EchoClientTest.java

```diff
@@ -0,0 +1,27 @@
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.net.Socket;
+
+import static org.junit.Assert.*;
+
+public class EchoClientTest {
+    @Test
+    public void fromStdInAndBack() {
```

Comment:
I'd make the test name a bit more descriptive. Something like `sent message is echoed back` maybe?